### PR TITLE
NATIVE-613: Target API level 29 (due 02.11.2020)

### PR DESCRIPTION
**Pull request by maxammann**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

**Comment by klinzo** [comment]
Already ready for review?

**Comment by maxammann** [comment]
As the review of  NATIVE-629 looks fine I'll open this now :)

File: android/app/build.gradle [hunk 1]
```diff
@@ -15,9 +15,14 @@ import com.android.build.OutputFile
  *   // the name of the generated asset file containing your JS bundle
  *   bundleAssetName: "index.android.bundle",
  *
- *   // the entry file for bundle generation
+ *   // the entry file for bundle generation. If none specified and
+ *   // "index.android.js" exists, it will be used. Otherwise "index.js" is
+ *   // default. Can be overridden with ENTRY_FILE environment variable.
  *   entryFile: "index.android.js",
  *
+ *   // https://reactnative.dev/docs/performance#enable-the-ram-format
```
Comment (klinzo) [inline android/app/build.gradle:23]: The `#enable-the-ram-format` does not exist, does it?

File: android/app/src/debug/java/tuerantuer/app/integreat/ReactNativeFlipper.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/**
```
Comment (klinzo) [inline android/app/src/debug/java/tuerantuer/app/integreat/ReactNativeFlipper.java:1]: Review can be skipped here right?

File: android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -17,7 +17,8 @@
     <activity
             android:name=".MainActivity"
             android:label="@string/BUILD_CONFIG_APP_NAME"
-            android:configChanges="keyboard|keyboardHidden|orientation|screenSize|layoutDirection|locale"
+            android:configChanges="keyboard|keyboardHidden|orientation|screenSize|layoutDirection|locale|uiMode"
+            android:launchMode="singleTask"
```
Comment (klinzo) [inline android/app/src/main/AndroidManifest.xml:21]: What is the reason for this?

File: android/build.gradle [hunk 4]
```diff
@@ -2,17 +2,18 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "28.0.3"
+        buildToolsVersion = "29.0.2"
         minSdkVersion = 16
-        compileSdkVersion = 28
-        targetSdkVersion = 28
+        compileSdkVersion = 29
+        targetSdkVersion = 29
```
Comment (klinzo) [inline android/build.gradle:8]: :taco: :tada:

File: android/build.gradle [hunk 5]
```diff
@@ -2,17 +2,18 @@
...
+        kotlinVersion = "1.3.50" // TODO: Remove in NATIVE-490
```
Comment (klinzo) [inline android/build.gradle:9]: Could you add a note to NATIVE-490? Why should it be removed in that issue?

File: docs/react-native-upgrades.md [hunk 6]
```diff
@@ -1,13 +1,29 @@
 # Upgrades
 
+## From 0.61.5 to 0.63.2
+
+This upgrade was needed in order to get to API level 29.
```
Comment (klinzo) [inline docs/react-native-upgrades.md:5]:
```suggestion
This upgrade was needed in order to get to android API level 29.
```

File: ios/Integreat.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -484,19 +485,13 @@
 				CODE_SIGN_STYLE = Manual;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_TEAM = 7272KE28TJ;
-				ENABLE_BITCODE = YES;
+				ENABLE_BITCODE = NO;
```
Comment (Schedulaar) [inline ios/Integreat.xcodeproj/project.pbxproj:488]: Interesting, why?

File: docs/react-native-upgrades.md [hunk 8]
```diff
@@ -1,13 +1,29 @@
...
+The following patches were applied manually: [Online diff](https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.63.0)
+
+### What to do after upgrade?
+* Run `./gradlew clean` in `android/`
+* Run Clean in XCode
+
+### Noteworthy Features
```
Comment (klinzo) [inline docs/react-native-upgrades.md:12]: :+1:

File: package.json [hunk 9]
```diff
@@ -38,10 +38,12 @@
     "@react-native-community/async-storage": "^1.9.0",
     "@react-native-community/geolocation": "^2.0.2",
     "@react-native-community/netinfo": "^5.5.0",
+    "@react-native-community/progress-bar-android": "^1.0.3",
+    "@react-native-community/progress-view": "^1.1.1",
```
Comment (Schedulaar) [inline package.json:42]: Where do we need these?

File: ios/Integreat.xcodeproj/project.pbxproj (hunk 7 above)
Comment (klinzo) [inline ios/Integreat.xcodeproj/project.pbxproj:488]: Is this intentional? Why?

File: src/index.js [hunk 10]
```diff
@@ -1,13 +1,10 @@
...
 
 import * as React from 'react'
-import { AppRegistry, YellowBox } from 'react-native'
+import { AppRegistry } from 'react-native'
 import App from './modules/app/components/App'
 import 'moment/min/locales'
 
-// @see: https://github.com/facebook/metro/issues/287
-YellowBox.ignoreWarnings(['Require cycle:'])
-
```
Comment (Schedulaar) [inline src/index.js:10]: The issue is not resolved (or is it?). Why did you remove this?

File: src/modules/app/constants/buildConfig.js [hunk 11]
```diff
@@ -17,13 +17,13 @@ import malteIntroSearchIcon from '../../../../build-configs/malte/assets/Search.
...
-    events: string,
-    language: string,
-    offers: string,
-    search: string
+    events: number,
+    language: number,
+    offers: number,
+    search: number
   }
 |}
```
Comment (Schedulaar) [inline src/modules/app/constants/buildConfig.js:28]: hui

File: package.json (hunk 9 above)
Comment (klinzo) [inline package.json:42]: If these are just sub dependencies, why do we need them here?

File: src/modules/app/containers/__tests__/NavigatorContainer.spec.js [hunk 12]
```diff
@@ -40,105 +37,93 @@ class MockNavigator extends React.Component<{}> {
...
+    const store = mockStore({})
+    const appSettings = new AppSettings()
+    await appSettings.setContentLanguage('de')
+    await appSettings.setSelectedCity('augsburg')
+    await appSettings.setIntroShown()
+    const result = render(<Provider store={store}><NavigatorContainer /></Provider>)
+    await waitForExpect(() => {
+      expect(result.getByText('Dashboard')).toBeTruthy()
     })
   })
```
Comment (Schedulaar) [inline src/modules/app/containers/__tests__/NavigatorContainer.spec.js:128]: What about this one?

File: src/modules/layout/components/MaterialHeaderButtons.js [hunk 13]
```diff
@@ -21,13 +21,13 @@ const onOverflowMenuPress = (cancelButtonLabel: string) => ({ overflowButtonRef,
...
 
-const MaterialHeaderButtons = (props: { cancelLabel: string, children: React.Node, theme: ThemeType }) => {
-  const { cancelLabel, theme, ...otherProps } = props
+const MaterialHeaderButtons = (props: {| cancelLabel: string, children: React.Node, theme: ThemeType |}) => {
+  const { cancelLabel, theme } = props
   return (
     <HeaderButtons HeaderButtonComponent={MaterialHeaderButton}
                    OverflowIcon={<MaterialIcon name='more-vert' size={23} color={theme.colors.textColor} />}
                    onOverflowMenuPress={onOverflowMenuPress(cancelLabel)}
-                   {...otherProps}
```
Comment (Schedulaar) [inline src/modules/layout/components/MaterialHeaderButtons.js:30]: are you sure, this was never used?

File: src/modules/app/containers/__tests__/NavigatorContainer.spec.js [hunk 14]
```diff
@@ -40,105 +37,93 @@ class MockNavigator extends React.Component<{}> {
...
+let TestRenderer
+let render
+let Provider
+
 describe('NavigatorContainer', () => {
+  beforeEach(() => {
+    jest.resetModules()
+    TestRenderer = require('react-test-renderer')
+    render = require('@testing-library/react-native').render
+    Provider = require('react-redux').Provider
```
Comment (klinzo) [inline src/modules/app/containers/__tests__/NavigatorContainer.spec.js:49]: Why is this required? Maybe add a comment

**Review by Schedulaar — APPROVED** [review]

File: src/modules/app/containers/__tests__/NavigatorContainer.spec.js [hunk 15]
```diff
@@ -40,105 +37,93 @@ class MockNavigator extends React.Component<{}> {
...
-      await appSettings.setSelectedCity('augsburg')
-      await appSettings.setIntroShown()
-      const result = render(<Provider store={store}><NavigatorContainer /></Provider>)
-      await waitForExpect(() => {
-        expect(result.getByText('Dashboard')).toBeTruthy()
-      })
-    })
-  })
-
-  it('should render the LandingContainer if no city is selected and the intro has already been shown', async () => {
```
Comment (klinzo) [inline src/modules/app/containers/__tests__/NavigatorContainer.spec.js:129]: Why did you remove this test?

**Review by klinzo — APPROVED** [review]
Good job :muscle:

File: package.json (hunk 9 above)
Comment (maxammann) [reply package.json:42]: They are required peer dependencies

File: src/index.js (hunk 10 above)
Comment (maxammann) [reply src/index.js:10]: I think we should keep an eye on this. Also the API of YellowBox changed so not sure whether this still works.

I think most libraries just need an update to remove this warning. at least I did not see it anymore.

File: src/modules/app/constants/buildConfig.js (hunk 11 above)
Comment (maxammann) [reply src/modules/app/constants/buildConfig.js:28]: hua

File: src/modules/layout/components/MaterialHeaderButtons.js (hunk 13 above)
Comment (maxammann) [reply src/modules/layout/components/MaterialHeaderButtons.js:30]: Nope, I readded this in the 629 PR which will be merged at the same time

File: android/app/build.gradle (hunk 1 above)
Comment (maxammann) [reply android/app/build.gradle:23]: Hm not my fault :D

File: android/app/src/debug/java/tuerantuer/app/integreat/ReactNativeFlipper.java (hunk 2 above)
Comment (maxammann) [reply android/app/src/debug/java/tuerantuer/app/integreat/ReactNativeFlipper.java:1]: Yes

File: android/app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (maxammann) [reply android/app/src/main/AndroidManifest.xml:21]: I dont know really.

File: ios/Integreat.xcodeproj/project.pbxproj (hunk 7 above)
Comment (maxammann) [reply ios/Integreat.xcodeproj/project.pbxproj:488]: https://issues.tuerantuer.org/browse/NATIVE-628

Comment (maxammann) [reply ios/Integreat.xcodeproj/project.pbxproj:488]: https://issues.tuerantuer.org/browse/NATIVE-628

File: package.json (hunk 9 above)
Comment (maxammann) [reply package.json:42]: This has to be done according to react-native-pdf

Comment (maxammann) [reply package.json:42]: This has to be done according to react-native-pdf

File: src/modules/app/containers/__tests__/NavigatorContainer.spec.js (hunk 14 above)
Comment (maxammann) [reply src/modules/app/containers/__tests__/NavigatorContainer.spec.js:51]: added one
